lib/tic_tac_toe/board.ex

```diff
@@ -1,27 +1,87 @@
 defmodule TicTacToe.Board do
   alias TicTacToe.Board
+  alias TicTacToe.BoardInspect
+  alias TicTacToe.BoardUpdate
 
-  defstruct [:board_state, game_over: false]
+  defstruct [:board_state]
 
-  defimpl TicTacToe.BoardInspect do
+  defimpl BoardInspect do
     def get(%Board{board_state: board_state}, position) do
       Enum.at(board_state, position - 1)
     end
+
+    def size(%Board{board_state: board_state}) do
+      number_of_positions = length(board_state)
+      row_length = number_of_positions |> :math.sqrt() |> trunc()
+      {number_of_positions, row_length}
+    end
+
+    def outcome(%Board{board_state: board_state} = board) do
+      winner = winner(board)
+
+      status =
+        cond do
+          winner -> :win
+          draw?(board_state) -> :draw
+          true -> :in_progress
+        end
+
+      {status, winner}
+    end
+
+    defp rows(%Board{board_state: board_state} = board) do
+      {_, row_length} = size(board)
+      1..length(board_state) |> Enum.to_list() |> Enum.chunk_every(row_length)
+    end
+
+    defp columns(rows) do
+      rows |> Enum.zip() |> Enum.map(&Tuple.to_list/1)
```

Comment:
💯 I never think of zip, this is cool

Reply:
I have to thank google for introducing me to zip and showing me how to transpose a nested list

Reply:
+1  🆒